mr/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -133,4 +146,93 @@ public DecomposedPredicate decomposePredicate(JobConf jobConf, Deserializer dese
+    return (LocationProvider) get(configuration, InputFormatConfig.LOCATION_PROVIDER);
+  }
+
+  /**
+   * Returns the Table EncryptionManager serialized to the configuration.
+   * @param configuration The configuration used to get the data from
+   * @return The Table EncryptionManager object
+   */
+  public static EncryptionManager encryption(Configuration configuration) {
+    return (EncryptionManager) get(configuration, InputFormatConfig.ENCRYPTION_MANAGER);
+  }
+
+  /**
+   * Returns the Table Schema serialized to the configuration.
+   * @param configuration The configuration used to get the data from
+   * @return The Table Schema object
+   */
+  public static Schema schema(Configuration configuration) {
+    return SchemaParser.fromJson(configuration.get(InputFormatConfig.TABLE_SCHEMA));
+  }
+
+  /**
+   * Returns the Table PartitionSpec serialized to the configuration.
+   * @param configuration The configuration used to get the data from
+   * @return The Table PartitionSpec object
+   */
+  public static PartitionSpec spec(Configuration configuration) {
+    return PartitionSpecParser.fromJson(schema(configuration), configuration.get(InputFormatConfig.PARTITION_SPEC));
+  }
+
+  /**
+   * Stores the serializable table data in the configuration.
+   * Currently the following is handled:
+   * <ul>
+   *   <li>- Location</li>
+   *   <li>- Schema</li>
+   *   <li>- Partition specification</li>
+   *   <li>- FileIO for handling table files</li>
+   *   <li>- Location provider used for file generation</li>
+   *   <li>- Encryption manager for encryption handling</li>
+   * </ul>
+   * @param configuration The target configuration to store to
+   * @param table The table which we want to store to the configuration
+   */
+  @VisibleForTesting
+  static void put(Configuration configuration, Table table) {
+    configuration.set(InputFormatConfig.TABLE_LOCATION, table.location());
+    configuration.set(InputFormatConfig.TABLE_SCHEMA, SchemaParser.toJson(table.schema()));
+    configuration.set(InputFormatConfig.PARTITION_SPEC, PartitionSpecParser.toJson(table.spec()));
+
+    put(configuration, InputFormatConfig.FILE_IO, table.io());
+    put(configuration, InputFormatConfig.LOCATION_PROVIDER, table.locationProvider());
+    put(configuration, InputFormatConfig.ENCRYPTION_MANAGER, table.encryption());
+  }
+
+  private static void put(Configuration configuration, String key, Serializable object) {
+    try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
+         ObjectOutputStream oos = new ObjectOutputStream(baos)) {
+      oos.writeObject(object);
+      configuration.set(key, Base64.getEncoder().encodeToString(baos.toByteArray()));
+    } catch (IOException ioe) {
+      throw new RuntimeException(String.format("Error serializing %s to configuration", object), ioe);
+    }
+  }
+
+  private static Object get(Configuration configuration, String key) {
```

Comment:
Could this class delegate to `SerializationUtil.serializeToBase64` and `SerializationUtil.deserializeFromBase64`?

Reply:
Definitely!
Removed the methods for serializing/deserializing! :)